src/Finder/RemovableRedirectFinderInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\Finder;
+
+use Doctrine\Common\Collections\Collection;
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+
+interface RemovableRedirectFinderInterface
+{
+    /**
+     * Returns the redirects that have for source the destination of the given redirect
+     *
+     * If the redirect has multiple channels it does the operation for each channel
+     * Ie: There is a RedirectionPath with ['a -> b', 'b -> c'], both without channel, findNextRedirect->('c -> a');
+     * will return new ArrayCollection(['a -> b']);
+     * If all those redirect had multiple channels, this would result in :
+     *      new ArrayCollection(['a -> b (channel1)', 'a -> b (channel2)', 'a -> b (channel3)', ...])
+     *
+     * @return Collection|RedirectInterface[]
+     */
+    public function findNextRedirect(RedirectInterface $redirect): Collection;
```

Comment:
```suggestion
    public function __invoke(RedirectInterface $redirect): Collection;
```

Maybe should this class just be invokable?

Reply:
Never used __invoke to be honest, but if I understand what it does, I don't think that's a good idea. the class is `RemovableRedirectFinder`, so we might suppose there will be more method to find RemovableRedirect no ?

Reply:
Basically if the class name says it all and the interface only exposes one method, it is eligible to be refactored as an invokable class. And it has the added benefit that this service can be used as a callable